Splitter/EarlGreyWelcomeScreenTests/EarlGreyWelcomeScreenTests.swift

```diff
@@ -0,0 +1,273 @@
+                                                 with: toastLabel,
+                                                 conditionName: conditionName)
+        
+        GREYAssertTrue(appearedSuccesfully, reason: "Toast appeared with correct text")
+    }
+    
+    func testRegisteringWithInvalidConfirmationPasswordDisplaysLocalisedError() {
+        let email = createEmail(with: "\(#function)")
+        let localisedErrorMessage = localizedStringWith(key: "InvalidPasswordError")
+        let toastLabel = EarlGrey.select(elementWithMatcher: grey_accessibilityID(AccessID.toastLabel))
+        let assertion = grey_text(localisedErrorMessage)
+        let conditionName = "Wait for toast to appear with correct text"
+        
+        register(email: email,
+                 password: password,
+                 confirmationPassword: shortPassword)
+        
+        let appearedSuccesfully = waitForSuccess(of: assertion,
+                                                 with: toastLabel,
+                                                 conditionName: conditionName)
+        
+        GREYAssertTrue(appearedSuccesfully, reason: "Toast appeared with correct text")
+    }
+    
+    func testRegisteringWithDifferentPasswordsDisplaysLocalisedError() {
+        let email = createEmail(with: "\(#function)")
+        let localisedErrorMessage = localizedStringWith(key: "PasswordMismatchError")
+        let toastLabel = EarlGrey.select(elementWithMatcher: grey_accessibilityID(AccessID.toastLabel))
+        let assertion = grey_text(localisedErrorMessage)
+        let conditionName = "Wait for toast to appear with correct text"
+        
+        register(email: email,
+                 password: password,
+                 confirmationPassword: differentPassword)
+        
+        let appearedSuccesfully = waitForSuccess(of: assertion,
+                                                 with: toastLabel,
+                                                 conditionName: conditionName)
+        
+        GREYAssertTrue(appearedSuccesfully, reason: "Toast appeared with correct text")
+    }
+    
+    func testRegisteringWithUsedEmailDisplaysFirebaseError() {
+        let firebaseErrorMessage = "The email address is already in use by another account."
+        let assertion = grey_text(firebaseErrorMessage)
+        let toastLabel = EarlGrey.select(elementWithMatcher: grey_accessibilityID(AccessID.toastLabel))
+        let conditionName = "Wait for toast to appear"
+        
+        register(email: registeredEmail,
+                 password: password,
+                 confirmationPassword: password)
+        
+        let appearedSuccesfully = waitForSuccess(of: assertion,
+                                                 with: toastLabel,
+                                                 conditionName: conditionName)
+        
+        GREYAssertTrue(appearedSuccesfully, reason: "Toast appeared with correct text")
```

Comment:
When we were using Earl Grey at another client, we wrote this wrapper to make the tests a little more succicnt. Perhaps you might want to take a look at it in other PRs - I'd be interested to see how you find it!

https://github.com/novoda/ios-demos/blob/master/Earl-Grey-Demo/Earl-Grey-DemoTests/UITest.swift

Reply:
Definitely 👍